numba_dpex/experimental/launcher.py

```diff
@@ -403,4 +584,9 @@ def call_kernel(kernel_fn, index_space, *kernel_args):
         kernel_args : List of objects that are passed to the numba_dpex.kernel
         decorated function.
     """
-    _launch_trampoline(kernel_fn, index_space, *kernel_args)
+    device_event = _submit_kernel(  # pylint: disable=E1120
+        kernel_fn,
+        index_space,
+        kernel_args,
+    )
+    device_event.wait()  # pylint: disable=E1101
```

Comment:
For this case, how will the `device_event` object get cleaned up? Previosuly, we had `sycl.dpctl_event_delete(self._builder, eref)`.

Reply:
`wrap_event_reference` allocates meminfo and sycl event object becomes smart pointer. Destructor of meminfo will call `DPCTLEvent_Delete` which is the same function called by `dpctl_event_delete` https://github.com/IntelPython/numba-dpex/blob/main/numba_dpex/core/runtime/_eventstruct.c#L19

Reply:
Here is the log of async call:
```
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x561537ebfc00 external_allocator=0x561537e91fb0 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x561537b7c1c0 external_allocator=0x561537df11b0 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x561537b49c50 external_allocator=0x561537fb8120 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: creating new dpctl event meminfo.
DPEXRT-DEBUG: scheduling nrt meminfo release.
DPEXRT-DEBUG: acquired meminfo.
DPEXRT-DEBUG: creating new dpctl event meminfo.
DPEXRT-DEBUG: creating new event object.
DPEXRT-DEBUG: released meminfo from host_task.
DPEXRT-DEBUG: deleting dpctl event reference.
DPEXRT-DEBUG: creating new event object.
DPEXRT-DEBUG: deleting dpctl event reference.
DPEXRT-DEBUG: released meminfo from host_task.
DPEXRT-DEBUG: released meminfo from host_task.
```

both dpctl event reference getting deleted.

And for `call_kernel`:
```
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x55f94f0452d0 external_allocator=0x55f94f4a6260 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x55f94f3f3470 external_allocator=0x55f94f403860 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: In DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 805
DPEXRT-DEBUG: usm type = 1 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 282.
DPEXRT-DEBUG: NRT_MemInfo_init mi=0x55f94f044c40 external_allocator=0x55f94f0e5b30 at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 476
DPEXRT-DEBUG: Done with unboxing call to DPEXRT_sycl_usm_ndarray_from_python at /home/yevhenii/Projects/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c, line 885
DPEXRT-DEBUG: creating new dpctl event meminfo.
DPEXRT-DEBUG: deleting dpctl event reference.
```
As you can see - it is getting deleted.